src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/features/instantiate/containers/index.tsx

```diff
@@ -33,35 +27,36 @@ export interface IPartyValidation {
   validParties: IParty[];
 }
 
-function InstantiateContainer() {
+const InstantiateContainer = () => {
   changeBodyBackground(AltinnAppTheme.altinnPalette.primary.greyLight);
-  const { org, app } = window as Window as IAltinnWindow;
 
   const [instantiating, setInstantiating] = React.useState(false);
-  const instantiation = useAppSelector(state => state.instantiation);
-  const selectedParty = useAppSelector(state => state.party.selectedParty);
-  const titleText: any = useAppSelector(state => {
-    const text = getTextFromAppOrDefault(titleKey, state.textResources.resources, state.language.language, [], true);
+  const instantiation = useAppSelector((state) => state.instantiation);
+  const selectedParty = useAppSelector((state) => state.party.selectedParty);
+  const titleText = useAppSelector((state) => {
+    const text = getTextFromAppOrDefault(
+      titleKey,
+      state.textResources.resources,
+      state.language.language,
+      [],
+      true,
+    );
     return text === titleKey ? '' : text;
   });
 
-  const createNewInstance = () => {
-    if (!selectedParty) {
-      return;
-    }
-    setInstantiating(true);
-    InstantiationActions.instantiate(org, app);
-  };
-
   React.useEffect(() => {
-    if (
-      !instantiating &&
-      !instantiation.instanceId
-    ) {
+    if (!instantiating && !instantiation.instanceId) {
+      const createNewInstance = () => {
+        if (selectedParty) {
+          const { org, app } = window as Window as IAltinnWindow;
+          setInstantiating(true);
+          InstantiationActions.instantiate(org, app);
+        }
+      };
+
       createNewInstance();
```

Comment:
```suggestion
    if (!instantiating && !instantiation.instanceId && selectedParty) {
      // create new instance
      const { org, app } = window as Window as IAltinnWindow;
      setInstantiating(true);
      InstantiationActions.instantiate(org, app);
    }
```

Reply:
Thanks for the suggestion :) Moved the `selectedParty` check to the outer if. I prefer to keep the function with a descriptive name instead of a comment to explain what is going on, as comments tend to get left behind during refactors later on.

Reply:
I can remove the function altogether if you think that makes it clearer? The function is really redundant here anyway.

Reply:
I definetly think the comment is clearer than the inline function. If the function could be defined outside the useEffect call (and not capture variables) it would be a different story.

Anyway it's probably just me being grumpy because I made a wrong comment on another issue because I was confused trying to follow the execution.

Reply:
I've updated it and removed the redundant function, and made a descriptive constant for the conditions check instead. That way we convey the meaning and improve readabilty without the need for a comment.